src/bessel.jl

```diff
@@ -758,3 +758,41 @@ External links: [DLMF](https://dlmf.nist.gov/10.2.6), [Wikipedia](https://en.wik
 See also: [`hankelh2(nu,x)`](@ref SpecialFunctions.hankelh2)
 """
 hankelh2x(nu, z) = besselhx(nu, 2, z)
+
+    
+ # In future just use `fastabs` from Base.Math
+ # https://github.com/JuliaLang/julia/blob/93fb785831dcfcc442f82fab8746f0244c5274ae/base/special/trig.jl#L1057
+function fastabs(x)
+    @static if isdefined(Base.Math, :fastabs)
+        return Base.Math.fastabs(x)
+    else
+       return abs(real(x)) + abs(imag(x))  
+    end
+end
+
+"""
+    jinc(x)
+
+Bessel function of the first kind divided by `x`.
+Following convention: ``\\operatorname{jinc}{x} = \\frac{2 \\cdot J_1{\\pi x}}{\\pi x}``.
+Sometimes known as sombrero or besinc function.
+
+External links: [Wikipedia](https://en.wikipedia.org/wiki/Sombrero_function)
+"""
+jinc(x::Number) = iszero(x) ? oftype(x, 1) : _jinc(float(x))
```

Comment:
This is wrong — the oftype conversion should happen *after* `float(x)`.  That is, this test should happen in `_jinc`, not in `jinc`

Reply:
And you can just use `one(x)` instead of `oftype(x,1)`.

Reply:
I'm not sure if I understand you correctly, but: `one(x)` should happen after float to preserve type stability? 
Currently it is `typeof(jinc(0)) != typeof(jinc(1))` and you want to have the same type, right?

Reply:
Yes.  That is, you should do
```jl
jinc(x::Number) = _jinc(float(x))
_jinc(x::Number) = iszero(x) ? one(x) : _jinc_core(x)
```